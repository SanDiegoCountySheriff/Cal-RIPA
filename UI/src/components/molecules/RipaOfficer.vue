<template>
  <div class="ripa-officer tw-pb-4">
    <ripa-form-header
      title="Officer Years of Experience"
      required
      subtitle="ยง999.226(a)(15)"
      class="tw-mb-4"
      v-on="$listeners"
    >
    </ripa-form-header>

    <v-container>
      <v-row no-gutters>
        <v-col cols="12" sm="12">
          <v-alert outlined dense type="info">
            <v-row align="center">
              <v-col class="grow">
                <template v-if="isValidUser">
                  {{ getOfficerInfo }}
                </template>
                <template v-if="!isValidUser">
                  <v-progress-circular
                    indeterminate
                    color="primary"
                  ></v-progress-circular>
                </template>
              </v-col>
              <v-col class="shrink">
                <v-btn color="primary" small @click="handleUpdateUser">
                  Update
                </v-btn>
              </v-col>
            </v-row>
          </v-alert>
        </v-col>
      </v-row>
    </v-container>
  </div>
</template>

<script>
import RipaFormHeader from '@/components/molecules/RipaFormHeader'
import RipaModelMixin from '@/components/mixins/RipaModelMixin'
import { OFFICER_ASSIGNMENTS } from '@/constants/form'

export default {
  name: 'ripa-officer',

  mixins: [RipaModelMixin],

  components: {
    RipaFormHeader,
  },

  data() {
    return {
      viewModelUser: this.user,
    }
  },

  inject: ['user'],

  computed: {
    isValidUser() {
      return this.user && this.user.assignment
    },

    getOfficerInfo() {
      if (this.user) {
        const otherType =
          this.user.assignment === 10 ? this.user.otherType : null
        const otherTypeText = otherType ? ` - ${otherType} - ` : ' - '
        return `${this.getOfficerAssignmentText()} ${otherTypeText} ${
          this.user.yearsExperience
        } Years`
      }

      return 'Officer information is missing. Please open user dialog and update.'
    },
  },

  methods: {
    getOfficerAssignmentText() {
      if (this.user && this.user.assignment) {
        const [assignment] = OFFICER_ASSIGNMENTS.filter(
          item => item.value === this.user.assignment,
        )
        return assignment.name
      }

      return ''
    },

    handleUpdateUser() {
      if (this.onUpdateUser) {
        this.onUpdateUser()
      }
    },
  },

  watch: {
    user(newValue) {
      this.viewModelUser = newValue
    },
  },

  props: {
    onUpdateUser: {
      type: Function,
      required: true,
    },
  },
}
</script>
