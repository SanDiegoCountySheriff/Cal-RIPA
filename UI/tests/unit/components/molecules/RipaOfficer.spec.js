import RipaOfficer from '@/components/molecules/RipaOfficer.vue'
import { shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Officer', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const factory = propsData => {
    return shallowMount(RipaOfficer, {
      vuetify,
      propsData: {
        ...propsData,
        onUpdateUser: jest.fn(),
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaOfficer, {
      vuetify,
      propsData: {
        onUpdateUser: jest.fn(),
      },
      provide: {
        user: () => {},
        stopVersion: () => 1,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
