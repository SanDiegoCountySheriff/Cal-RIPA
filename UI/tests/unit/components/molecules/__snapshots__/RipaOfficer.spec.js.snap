// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Officer should match snapshot 1`] = `
<div class="ripa-officer tw-pb-4" onupdateuser="function () {
        return fn.apply(this, arguments);
      }">
  <div class="ripa-form-header tw-mb-4">
    <div class="row tw-ml-1 tw-mt-1"><span class="v-tooltip v-tooltip--top"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-alert theme--light error--text" style="font-size: 16px;"></i></span> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
      <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Officer Years of Experience</span></div>
    </div>
  </div>
  <div class="container">
    <div class="row no-gutters">
      <div class="col-sm-12 col-12">
        <div role="alert" class="v-alert v-sheet v-sheet--outlined theme--light v-alert--dense v-alert--outlined info--text">
          <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon mdi mdi-information theme--light info--text"></i>
            <div class="v-alert__content">
              <div class="row align-center">
                <div class="grow col">
                  <!---->
                  <div role="progressbar" aria-valuemin="0" aria-valuemax="100" class="v-progress-circular v-progress-circular--visible v-progress-circular--indeterminate primary--text" style="height: 32px; width: 32px;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="22.857142857142858 22.857142857142858 45.714285714285715 45.714285714285715" style="transform: rotate(0deg);">
                      <circle fill="transparent" cx="45.714285714285715" cy="45.714285714285715" r="20" stroke-width="5.714285714285714" stroke-dasharray="125.664" stroke-dashoffset="125.66370614359172px" class="v-progress-circular__overlay"></circle>
                    </svg>
                    <div class="v-progress-circular__info"></div>
                  </div>
                </div>
                <div class="shrink col"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small primary"><span class="v-btn__content">
                Update
              </span></button></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
